presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -775,6 +795,64 @@ private void queryCompletedEvent(Optional<ExecutionFailureInfo> failureInfo)
             queryInfoOutputPath.ifPresent(path -> writeQueryInfo(path, queryInfo, queryInfoJsonCodec));
         }
 
+        private void processShuffleStats()
+        {
+            List<PrestoSparkShuffleStats> statsList = new ArrayList<>(shuffleStatsCollector.value());
+            Map<ShuffleStatsKey, List<PrestoSparkShuffleStats>> statsMap = new HashMap<>();
+            Set<Integer> fragments = new HashSet<>();
+            for (PrestoSparkShuffleStats stats : statsList) {
+                ShuffleStatsKey key = new ShuffleStatsKey(stats.getFragmentId(), stats.getOperation());
+                statsMap.computeIfAbsent(key, (ignored) -> new ArrayList<>()).add(stats);
+                fragments.add(stats.getFragmentId());
+            }
+            List<Integer> sortedFragments = new ArrayList<>(fragments);
```

Comment:
Another way to do this is to make `statsMap` a `TreeMap` so the result can be traversed in a ordered way 😄 . You can keep the current way.

Reply:
Yeah, the code with TreeMap is much simpler =) My bad =)